.github/workflows/on-pull-request-closed.yml

```diff
@@ -28,4 +35,12 @@ jobs:
           aws configure set aws_access_key_id $CLOUDFLARE_R2_ACCESS_KEY_ID
           aws configure set aws_secret_access_key $CLOUDFLARE_R2_SECRET_ACCESS_KEY
       - name: Remove artifacts
-        run: aws s3 rm s3://${{ vars.CLOUDFLARE_R2_BUCKET_NAME }}/${{ github.head_ref }} --endpoint-url ${{ vars.CLOUDFLARE_R2_ENDPOINT }} --recursive
+        run: |
+          BRANCH="${{ github.head_ref }}"
+
+          if [[ ! $BRANCH =~ ^[a-zA-Z0-9_\-\/]+$ ]]; then
+            echo "Invalid branch name"
+            exit 1
```

Comment:
Guards against malicious branch names, such as ones that contain scripting commands.